healthchain/data_generator/data_generator.py

```diff
@@ -48,55 +48,53 @@ def set_workflow(self, workflow: str):
         self.workflow = workflow
 
     def generate(
-        self, constraints: Optional[list] = None, free_text_json: Optional[str] = None
+        self,
+        constraints: Optional[list] = None,
+        free_text_path: Optional[str] = None,
+        column_name: Optional[str] = None,
     ) -> BaseModel:
         results = []
 
         if self.workflow not in self.mappings.keys():
             raise ValueError(f"Workflow {self.workflow} not found in mappings")
 
-        if free_text_json is not None:
-            parsed_free_text = self.free_text_parser(free_text_json)
-        else:
-            parsed_free_text = {self.workflow.value: []}
-
         for resource in self.mappings[self.workflow]:
             generator_name = resource["generator"]
             generator = self.fetch_generator(generator_name)
             result = generator.generate(constraints=constraints)
 
             results.append(Bundle_EntryModel(resource=result))
 
-        if (
-            self.workflow.value in parsed_free_text.keys()
-            and parsed_free_text[self.workflow.value]
-        ):
-            results.append(
-                Bundle_EntryModel(
-                    resource=random.choice(parsed_free_text[self.workflow.value])
-                )
-            )
+        parsed_free_text = (
+            self.free_text_parser(free_text_path, column_name)
+            if free_text_path
+            else None
+        )
+        if parsed_free_text:
+            results.append(Bundle_EntryModel(resource=random.choice(parsed_free_text)))
         output = OutputDataModel(context={}, resources=BundleModel(entry=results))
         self.data = output
         return output
 
-    def free_text_parser(self, free_text: str) -> dict:
-        with open(free_text) as f:
-            free_text = json.load(f)
+    def free_text_parser(self, path_to_csv: str, column_name: str) -> dict:
+        column_data = []
+        with open(path_to_csv, mode="r", newline="") as file:
```

Comment:
you might want to use `pathlib` here to check if it's a valid path and if it exists in a try..catch block